File: two-sum_1.js
const arr = [3, 2, 3];
const target = 5;

const twoSum = (arr, target) => {
  for (let i = 0; i < arr.length; i++) {
    if (arr[i] + arr[i + 1] === target) 
    return [i, i + 1];

    for (let j = i + 1; j < arr.length; j++) {
      if (arr[i] + arr[j] === target)
      return [i, j];
    }
  }
};

const index = twoSum(arr, target);
console.log(index);

File: two-sum_1.py
nums = [2, 1, 5, 3]
target = 4

def twoSum(nums, target):
    new_dict = {}   # val: index

    for i, n in enumerate(nums):
        diff = target - n
        if diff in new_dict:
            return [new_dict[diff], i]
        new_dict[n] = i
    
    return

print(twoSum(nums, target))


# Time complexity: O(n)
# Space complexity: O(n)